qubesadmin/app.py

```diff
@@ -658,20 +681,19 @@ def qubesd_call(self, dest, method, arg=None, payload=None,
 
         .. warning:: *payload_stream* will get closed by this function
         """
-        if payload and payload_stream:
-            raise ValueError(
-                'Only one of payload and payload_stream can be used')
         service_name = method
         if arg is not None:
             service_name += '+' + arg
         p = subprocess.Popen([qubesadmin.config.QREXEC_CLIENT_VM,
                               dest, service_name],
-                             stdin=(payload_stream or subprocess.PIPE),
+                             stdin=subprocess.PIPE,
```

Comment:
Shouldn't this still connect `payload_stream` directly if no `payload` is set?

Reply:
Okay. That does make the code more complicated (3 possible paths, not 2), but runs a bit faster and potentially avoids the deadlock in that case. Refactored.